pylint/checkers/typecheck.py

```diff
@@ -1921,6 +1930,23 @@ def _includes_version_compatible_overload(self, attrs: list):
         )
         return not is_py310_builtin or self._py310_plus
 
+    def _recursive_search_for_classdef_type(
+        self, node: nodes.ClassDef, operation: Literal["__or__", "__ror__"]
+    ) -> bool | VERSION_COMPATIBLE_OVERLOAD:
+        if not isinstance(node, nodes.ClassDef):
+            return False
+        for node_or_ancestor in itertools.chain((node,), node.ancestors()):
+            if is_classdef_type(node_or_ancestor):
+                node = node_or_ancestor
+            try:
+                attrs = node.getattr(operation)
+                if self._includes_version_compatible_overload(attrs):
+                    return VERSION_COMPATIBLE_OVERLOAD_SENTINEL
+                return True
+            except astroid.NotFoundError:
+                return True
+        return False
```

Comment:
It seems we never reach this line during the tests, but I don't know if the test case I suggested would work.

Reply:
Mypy insisted on it, but it's unreachable, since the iterator cannot be empty and every path through the iterator has a return statement. I'll add a pragma.